File: rb/features.rb
# XenoCanto SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module XenoCantoFeatures
  def self.make_feature(name)
    case name
    when "base"
      XenoCantoBaseFeature.new
    when "ratelimit"
      XenoCantoRatelimitFeature.new
    when "retry"
      XenoCantoRetryFeature.new
    when "test"
      XenoCantoTestFeature.new
    when "timeout"
      XenoCantoTimeoutFeature.new
    else
      XenoCantoBaseFeature.new
    end
  end
end

File: py/features.py
# XenoCanto SDK feature factory

from feature.base_feature import XenoCantoBaseFeature
from feature.test_feature import XenoCantoTestFeature


def _make_feature(name):
    features = {
        "base": lambda: XenoCantoBaseFeature(),
        "test": lambda: XenoCantoTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
